### PR TITLE
chore: Set upgrade heights and change codename

### PR DESCRIPTION
**Pull request by rjan90**
## Proposed Changes
- Set calibration height to `1390534` which is `2024-02-27T14:00:00Z`
- Set upgrade epoch in Butterfly to epoch `5760` which is two days from reset, allowing for us to get some states in the network before the upgrade height.
- Change nv22 codename from Pineapple to Dragon, and Mango to Phoenix (Drand-switch)

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Review by arajasek — APPROVED** [review]

File: build/params_2k.go [hunk 1]
```diff
@@ -67,9 +67,9 @@ var UpgradeThunderHeight = abi.ChainEpoch(-23)
 
 var UpgradeWatermelonHeight = abi.ChainEpoch(-24)
 
-var UpgradePineappleHeight = abi.ChainEpoch(20)
+var UpgradeDragonHeight = abi.ChainEpoch(20)
 
-var UpgradeMangoHeight = UpgradePineappleHeight + 10
+var UpgradeMangoHeight = UpgradeDragonHeight + 10
```
Comment (arajasek) [inline build/params_2k.go:72]: Per discussion, we'll probably wind up going with something like `+ 100` instead. But that doesn't have to change in this PR.

Comment (rjan90) [reply build/params_2k.go]: Do we also want the Drand switch to be `+100` epochs away from the nv22 upgrade in the butterfly/calibration/mainnet ?

File: api/types.go [hunk 2]
```diff
@@ -349,7 +349,7 @@ type ForkUpgradeParams struct {
 	UpgradeLightningHeight   abi.ChainEpoch
 	UpgradeThunderHeight     abi.ChainEpoch
 	UpgradeWatermelonHeight  abi.ChainEpoch
-	UpgradePineappleHeight   abi.ChainEpoch
+	UpgradeDragonHeight      abi.ChainEpoch
```
Comment (Stebalien) [inline api/types.go:352]: Mango as well?

File: build/params_2k.go [hunk 3]
```diff
@@ -67,9 +67,9 @@ var UpgradeThunderHeight = abi.ChainEpoch(-23)
 
 var UpgradeWatermelonHeight = abi.ChainEpoch(-24)
 
-var UpgradePineappleHeight = abi.ChainEpoch(20)
+var UpgradeDragonHeight = abi.ChainEpoch(20)
 
-var UpgradeMangoHeight = UpgradePineappleHeight + 10
+var UpgradeMangoHeight = UpgradeDragonHeight + 100
```
Comment (Stebalien) [inline build/params_2k.go:72]: - Knight
- Serpent
- Slayer
- ...

Pinapple -> mango, dragon -> mango???? no.

File: api/types.go (hunk 2 above)
Comment (Stebalien) [reply api/types.go:352]: Or are we saying that this hight isn't configurable?

File: build/params_calibnet.go [hunk 4]
```diff
@@ -89,10 +89,10 @@ const UpgradeWatermelonFixHeight = 1070494
 // 2023-11-21T13:00:00Z
 const UpgradeWatermelonFix2Height = 1108174
 
-// ??????
-const UpgradePineappleHeight = 999999999999999
+// 2024-02-27T14:00:00Z
```
Comment (Stebalien) [inline build/params_calibnet.go:92]: Verified.

**Review by Stebalien — APPROVED** [review]
Quibbles over names but this looks correct.

File: build/params_2k.go (hunk 1 above)
Comment (Stebalien) [reply build/params_2k.go]: > Do we also want the Drand switch to be +100 epochs away from the nv22 upgrade in the butterfly/calibration/mainnet ?

We really _only_ care about mainnet.

@arajasek can we make this 120 so it falls on an hour mark?

File: build/params_2k.go (hunk 3 above)
Comment (rjan90) [reply build/params_2k.go]: Lollol, I hear Dragons like Mango??! Updated Mango to Phoenix in: https://github.com/filecoin-project/lotus/pull/11599/commits/cea5ae65af719c905dd4ffe629b5ebfad1962815

File: api/types.go (hunk 2 above)
Comment (arajasek) [reply api/types.go:352]: We should still have it, it's not so much about configurability as information about the network.

**Review by Stebalien — APPROVED** [review]

**Pull request merged**